llvm/lib/Transforms/Scalar/IndVarSimplify.cpp

```diff
@@ -1931,6 +1932,30 @@ bool IndVarSimplify::run(Loop *L) {
     }
   }
 
+  // The loop exit values have been updated; insert the debug location
+  // for the induction variable with its final value.
```

Comment:
Why is this code not inside rewriteLoopExitValues, which does the actual transform?

Reply:
Moved to `rewriteLoopExitValues`.